libraries/standard/mqtt/src/mqtt.c

```diff
@@ -345,8 +378,12 @@ static int32_t recvExact( const MQTTContext_t * pContext,
             receiveError = true;
         }
 
-        if( ( bytesRemaining > 0U ) &&
-            ( calculateElapsedTime( getTimeStampMs(), entryTimeMs ) > timeoutMs ) )
+        if( getTimeStampMs != NULL )
+        {
+            elapsedTime = calculateElapsedTime( getTimeStampMs(), entryTimeMs );
+        }
+
+        if( ( bytesRemaining > 0U ) && ( elapsedTime >= timeoutMs ) )
```

Comment:
This check was `>` earlier which is changed to `>='. Is it intentional? What is the expected behavior if getTimeStampMs is NULL.

Reply:
Yes, if `getTimeStampMs` is NULL, then both `elapsedTime` and `timeoutMs` here are expected to be 0. Changing the comparison to `>=` ensures that this will run for a single iteration in that case. I will add a comment to clarify that